AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m

```diff
@@ -0,0 +1,135 @@
+      [[NSUserDefaults standardUserDefaults] setObject:value forKey:newKey];
+      [[NSUserDefaults standardUserDefaults] removeObjectForKey:oldKey];
+      MSLogVerbose([MSAppCenter logTag], @"Migrating the key %@ -> %@", oldKey, newKey);
+    }
+  }
+  [self setObject:@(1) forKey:serviceHasMigratedKey];
+}
+
+- (NSString *)getAppCenterKeyFrom:(NSString *)key {
+    if ([key hasPrefix:kMSUserDefaultsPrefix]) {
+        return key;
+    } else {
+        return [kMSUserDefaultsPrefix stringByAppendingString:key];
+    }
+}
+
+- (id)objectForKey:(NSString *)key {
+  NSString *keyPrefixed = [self getAppCenterKeyFrom:key];
+  return [[NSUserDefaults standardUserDefaults] objectForKey:keyPrefixed];
+}
+
+- (void)setObject:(id)o forKey:(NSString *)key {
+  NSString *keyPrefixed = [self getAppCenterKeyFrom:key];
+  [[NSUserDefaults standardUserDefaults] setObject:o forKey:keyPrefixed];
+}
+
+- (void)removeObjectForKey:(NSString *)key {
+  NSString *keyPrefixed = [self getAppCenterKeyFrom:key];
+  [[NSUserDefaults standardUserDefaults] removeObjectForKey:keyPrefixed];
+}
+
+- (NSDictionary *)updateDictionary:(NSDictionary *)dict forKey:(NSString *)key expiration:(float)expiration {
+  NSString *keyPrefixed = [self getAppCenterKeyFrom:key];
+  NSMutableDictionary *update = [[NSMutableDictionary alloc] initWithDictionary:dict];
+
+  // Get from local store.
+  NSDictionary *store = [[NSUserDefaults standardUserDefaults] dictionaryForKey:keyPrefixed];
+
+  CFAbsoluteTime ts = [(NSNumber *)store[kMSUserDefaultsTs] doubleValue];
+  MSLogVerbose([MSAppCenter logTag], @"Settings:store[%@]=%@", keyPrefixed, store);
+
+  // Force update if timestamp expiration is reached.
+  if (ts <= 0.0 || expiration <= 0.0f || fabs(CFAbsoluteTimeGetCurrent() - ts) < (double)expiration) {
+
+    // Remove if already in store and value is the same.
+    for (NSString *k in [store allKeys]) {
+      if (update[k] != nil && [(NSObject *)update[k] isEqual:store[k]])
+        [update removeObjectForKey:k];
+    }
+  }
+
+  // If still values to update.
+  if ([update count] > 0) {
+    MSLogDebug([MSAppCenter logTag], @"Settings:update[%@]=%@", keyPrefixed, update);
+
+    // Copy store as a mutable version.
+    NSMutableDictionary *d = [store mutableCopy];
+    if (d == nil)
+      d = [[NSMutableDictionary alloc] initWithCapacity:[update count]];
+
+    // Append update to the current store.
+    [d addEntriesFromDictionary:update];
+
+    // Set new timestamp.
+    d[kMSUserDefaultsTs] = @(CFAbsoluteTimeGetCurrent());
+
+    // Save.
+    [[NSUserDefaults standardUserDefaults] setObject:d forKey:keyPrefixed];
+  }
+
+  return update;
+}
+
+- (NSDictionary *)updateDictionary:(NSDictionary *)dict forKey:(NSString *)key {
+  return [self updateDictionary:dict forKey:key expiration:0.0];
+}
+
+- (BOOL)updateObject:(id)o forKey:(NSString *)key expiration:(float)expiration {
+  NSDictionary *update = [self updateDictionary:@{@"v" : o} forKey:key expiration:expiration];
```

Comment:
v?

Reply:
I don't think yet another migration would be a good idea...